### ISSUE TITLE
Error when importing after successful installation (MacOS 11.6)

### ISSUE THREAD

**Issue by pzelasko** [csukuangfj/kaldilm#12]
```
(lhotse) ➜  ASR git:(master) ✗ pip install kaldilm            
Collecting kaldilm
  Using cached kaldilm-1.8.tar.gz (48 kB)
Building wheels for collected packages: kaldilm
  Building wheel for kaldilm (setup.py) ... done
  Created wheel for kaldilm: filename=kaldilm-1.8-cp38-cp38-macosx_11_0_x86_64.whl size=90182 sha256=23d18b84186b3eef9fe49a9aedad4e0a4296914ad5bdf4776b26b512221fbb87
  Stored in directory: /Users/pzelasko/Library/Caches/pip/wheels/85/7d/63/f2dd586369b8797cb36d213bf3a84a789eeb92db93d2e723c9
Successfully built kaldilm
Installing collected packages: kaldilm
Successfully installed kaldilm-1.8

(lhotse) ➜  ASR git:(master) ✗ python             
Python 3.8.10 (default, May 19 2021, 11:01:55) 
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import kaldilm
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/pzelasko/miniconda3/envs/lhotse/lib/python3.8/site-packages/kaldilm/__init__.py", line 1, in <module>
    from .arpa2fst import arpa2fst
  File "/Users/pzelasko/miniconda3/envs/lhotse/lib/python3.8/site-packages/kaldilm/arpa2fst.py", line 3, in <module>
    import _kaldilm
ImportError: dlopen(/Users/pzelasko/miniconda3/envs/lhotse/lib/python3.8/site-packages/_kaldilm.cpython-38-darwin.so, 2): Library not loaded: @rpath/libkaldilm_core.dylib
  Referenced from: /Users/pzelasko/miniconda3/envs/lhotse/lib/python3.8/site-packages//_kaldilm.cpython-38-darwin.so
  Reason: image not found
```

Note that this part is interesting: `Library not loaded: @rpath/libkaldilm_core.dylib`

I searched for this library with `find $path_to_conda_env_dir -name "libkaldilm*"` and haven't found it.

**Comment by csukuangfj** [comment]
Did you try

conda install -c k2-fsa -c conda-forge kaldilm

---

I will look into your error.

**Comment by csukuangfj** [comment]
```bash
pip install kaldilm==1.10
```
should work now.

It is fixed in #13 .

**Comment by csukuangfj** [comment]
Note:
you may need to run
```
export DYLD_LIBRARY_PATH=/Users/pzelasko/miniconda3/envs/lhotse/lib/python3.8/site-packages:$DYLD_LIBRARY_PATH
```

**Comment by pzelasko** [comment]
TL;DR it helped, but your conda solution didn't work, and I found a new quirk on MacOS unrelated to this project.

---

This command:
```
conda install -c k2-fsa -c conda-forge kaldilm
```

fails on MacOS with the following message:

```
$ conda install -c k2-fsa -c conda-forge kaldilm
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - kaldilm

Current channels:

  - https://conda.anaconda.org/k2-fsa/osx-64
  - https://conda.anaconda.org/k2-fsa/noarch
  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

---

I had quite some debugging with kaldilm==1.10 from pip -- long story short, it turns out when you run `$ ./prepare.sh`, `DYLD_LIBRARY_PATH` is somehow unset; I added `echo $DYLD_LIBRARY_PATH` to the script and this is what I'm getting:

```
$ ./prepare.sh --stage 4
$ echo $DYLD_LIBRARY_PATH
/Users/pzelasko/miniconda3/envs/lhotse//lib/python3.8/site-packages/
2021-10-08 22:23:07 (prepare.sh:24:main) dl_dir: /Users/pzelasko/jhu/icefall/egs/yesno/ASR/download
./prepare.sh: line 62: DYLD_LIBRARY_PATH: unbound variable
```

I have no idea why. It's not related to kaldilm in any way though.

**Comment by csukuangfj** [comment]
> but your conda solution didn't work

Sorry, there are only conda packages for Linux. I have not set it up for macOS yet.
For macOS, please use `pip install`.

----

> I have no idea why. It's not related to kaldilm in any way though.

Did you use `export DYLD_LIBRARY_PATH=xxx` or just use `DYLD_LIBRARY_PATH=xxx`?

I just tested it locally that if you run:
```
$ export DY_LD_LIBRARY_PATH=xxx
$ ./prepare.sh
(Inside prepare.sh, you use  echo "DYLD_LIBRARY_PATH: $DYLD_LIBRARY_PATH",
it will print the value of DYLD_LIBRARY_PATH)
``` 

I cannot reproduce what you are facing.

**Comment by pzelasko** [comment]
Gotcha. Might be something related to my setup then, as I did use `export`. Anyway, I can make it work when I define the variable inside the script. I’ll close this issue, thanks for help!
